modules/rostests/apitests/ntdll/NtAllocateVirtualMemory.c

```diff
@@ -561,8 +560,37 @@ CheckAdjacentVADs()
     BaseAddress = UlongToPtr(0x50010000);
     Size = 0x10000;
     Status = NtFreeVirtualMemory(NtCurrentProcess(), &BaseAddress, &Size, MEM_RELEASE);
+    // Result varies between ReactOS and Windows.
     ok_ntstatus(Status, STATUS_SUCCESS);
-
+    if (!NT_SUCCESS(Status))
+    { // Unexpected, cleanup.
+        trace("ReactOS does not support \"case B\" yet\n");
+        ok_ntstatus(Status, STATUS_FREE_VM_NOT_AT_BASE);
+        // Release the 3 64k region.
+        BaseAddress = UlongToPtr(0x50000000);
+        Size = 0x30000;
+        Status = NtFreeVirtualMemory(NtCurrentProcess(), &BaseAddress, &Size, MEM_RELEASE);
+        ok_ntstatus(Status, STATUS_SUCCESS);
+    }
+    else
+    {
+        trace("Windows supports \"case B\"\n");
```

Comment:
This distinction makes no sense, don't write tests for ROS.
If ROS fails, try to properly clean up and gracefully handle the failure -- ideally the result is simply a test failure. If you need to skip some tests, use skip().
If something crashes ROS, let's merge a fix for that crash before merging your test.